pkgs/by-name/bi/bitwarden-desktop/linux.nix

```diff
@@ -0,0 +1,159 @@
+{ lib, pname, meta, buildNpmPackage, cargo, copyDesktopItems, dbus, electron_28
+, fetchFromGitHub, glib, gnome, gtk3, jq, libsecret, makeDesktopItem
+, makeWrapper, moreutils, napi-rs-cli, nodejs_18, patchutils_0_4_2, pkg-config
+, python3, runCommand, rustc, rustPlatform, ... }:
+
+let
+  icon = "bitwarden";
+  electron = electron_28;
+in buildNpmPackage rec {
+  inherit pname meta;
+  version = "2024.3.0";
```

Comment:
The versions between Linux and Darwin are set independently at the moment.

Reply:
> Is there precedent for having the version vary by platform?

I based this work off of how Spotify is [currently packaged](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/audio/spotify). They set the versions independently.

> I think this is the big question here since the two are vastly different derivations: should this just be a separate bitwarden-desktop-bin?

Also here we can look at the Spotify package. There's only one package, where the Darwin version repackages the .dmg and the Linux one is built (very differently) from the Ubuntu Snap package. These OS-specific files have different maintainers. I would prefer to follow that format: have a single `bitwarden-desktop` package (because it is discoverable and means users with e.g. a home-manager configuration for both Darwin and Linux don't need to special-case) but with different maintainers.